src/main/groovy/org/transmartproject/batch/support/ScientificNotationFormat.groovy

```diff
@@ -0,0 +1,23 @@
+package org.transmartproject.batch.support
+
+import java.text.DecimalFormat
+import java.text.NumberFormat
+import java.text.ParsePosition
+
+/**
+ * Supports parsing of the scientific format numbers.
+ * Variations like 1.002e+9 and 1.002*10^9 are allowed.
+ */
+class ScientificNotationFormat extends NumberFormat {
+
+    @Delegate DecimalFormat decimalFormat = new DecimalFormat("0.###E0")
+
+    Number parse(String text, ParsePosition pos) {
+        String processedText = text
+                .replace('e', 'E')
+                .replace('E+', 'E')
```

Comment:
I see. Two problems. You need to force a locale, because the symbol `E` is locale-depend (see e.g. http://demo.icu-project.org/icu-bin/locexp?d_=en&_=ar_SA). Second, this is too lax. It will accept `'1aa'` as as number. You should look at the parse position and ensure that the whole string's been parsed.